src/interpreter/tests/wasm.rs

```diff
@@ -0,0 +1,116 @@
+//{Internal, External, Type, FunctionType, ValueType};
+
+use elements::deserialize_buffer;
+use elements::Module;
+use interpreter::EnvParams;
+use interpreter::ExecutionParams;
+
+use interpreter::module::ModuleInstanceInterface;
+use interpreter::program::ProgramInstance;
+use interpreter::value::RuntimeValue;
+
+// ///////////////////////////////////////////////////////////////////////////
+// In this test a WASM (generated from C++ below) is deserialized
+// and executed by the interpreter
+// ///////////////////////////////////////////////////////////////////////////
+
+//    /// @file inc.cpp
+//    #include <emscripten.h> // macro EMSCRIPTEN_KEEPALIVE
+//    #include <stdint.h>
+//    extern "C" {
+//    uint32_t EMSCRIPTEN_KEEPALIVE inc_i32(uint32_t param) {
+//        return ++param;
+//    }
+//    } // extern "C"
+//
+
+// ///////////////////////////////////////////////////////////////////////////
+// The corresponding WAST respresentation is below
+// ///////////////////////////////////////////////////////////////////////////
+
+//    (module
+//    (type $0 (func (param i32) (result i32)))
+//    (type $1 (func))
+//    (import "env" "memoryBase" (global $import$0 i32))
+//    (import "env" "memory" (memory $0 256))
+//    (import "env" "table" (table 0 anyfunc))
+//    (import "env" "tableBase" (global $import$3 i32))
+//    (global $global$0 (mut i32) (i32.const 0))
+//    (global $global$1 (mut i32) (i32.const 0))
+//    (export "_inc_i32" (func $0))
+//    (export "__post_instantiate" (func $2))
+//    (func $0 (type $0) (param $var$0 i32) (result i32)
+//    (i32.add
+//    (get_local $var$0)
+//    (i32.const 1)
+//    )
+//    )
+//    (func $1 (type $1)
+//    (nop)
+//    )
+//    (func $2 (type $1)
+//    (block $label$0
+//    (set_global $global$0
+//    (get_global $import$0)
+//    )
+//    (set_global $global$1
+//    (i32.add
+//    (get_global $global$0)
+//    (i32.const 5242880)
+//    )
+//    )
+//    (call $1)
+//    )
+//    )
+//    ;; custom section "dylink", size 5
+//    )
+//
+
+// Name of function contained in WASM byte sequence (note the trailing underline)
+const INC_FUNCTION_NAME : & 'static str = "_inc_i32";
+
+// The WASM code embedded as octet sequence
+const INC_FUNCTION_WASM :  &[u8] =
```

Comment:
Thanks!

Just a little not about raw data here. I think you'd better put wasm-file to [https://github.com/NikVolf/parity-wasm/tree/master/res/cases/v1](/res/v1/cases) and then

1.  and include it with [https://doc.rust-lang.org/std/macro.include_bytes.html](include_bytes!) at compile time

2. or just use [deserialize_file](https://github.com/NikVolf/parity-wasm/blob/master/src/elements/module.rs#L209) at runtime